components/charts/ColorGradientLegend.tsx

```diff
@@ -49,45 +51,81 @@ export default function ColorGradientLegend({
   const widthNoPadding = width - padding[1];
   const rectHeight = height - padding[0];
   return (
-    <svg className={className} height={height} width={width}>
-      <defs>
-        <linearGradient id="gradient-legend">
-          {colorRange.map((color, idx) => (
-            <stop key={idx} offset={idx / lastColorIdx} stopColor={String(color)} />
-          ))}
-        </linearGradient>
-      </defs>
-      <g transform={`translate(${padding[3]}, ${padding[0]})`}>
-        <rect fill="url(#gradient-legend)" height={rectHeight} width={widthNoPadding} x={0} y={0} />
-        {colorDomain.map((d, idx) => (
-          <text
-            className="gradient-legend-text"
-            dy="-2px"
-            key={idx}
-            x={widthNoPadding * (idx / lastDomainIdx)}
-            y={0}>
-            {d}
-          </text>
-        ))}
-      </g>
-      {colorNA && (
-        <g transform={`translate(${widthNoPadding}, ${padding[0]})`}>
-          <rect
-            fill={colorNA}
-            height={rectHeight}
-            width={padding[1] - NA_SPACING}
-            x={NA_SPACING}
-            y={0}
-          />
-          <text
-            className="gradient-legend-na"
-            dy="-2px"
-            x={NA_SPACING + (padding[1] - NA_SPACING) / 2}
-            y={0}>
-            N/A
-          </text>
-        </g>
+    <ThemeConsumer>
+      {({ colors }) => (
+        <svg className={className} height={height} width={width}>
+          <defs>
+            <linearGradient id="gradient-legend">
+              {colorRange.map((color, idx) => (
+                <stop key={idx} offset={idx / lastColorIdx} stopColor={String(color)} />
+              ))}
+            </linearGradient>
+
+            <pattern
+              id="stripes"
+              width="30"
+              height="30"
+              patternTransform="rotate(45 0 0)"
+              patternUnits="userSpaceOnUse">
+              {NA_GRADIENT_LINE_INCREMENTS.map((i) => (
+                <React.Fragment key={i}>
+                  <line
+                    x1={i}
+                    y1="0"
+                    x2={i}
+                    y2="30"
+                    style={{ stroke: colors.gray71, strokeWidth: NA_SPACING }}
+                  />
+                  <line
+                    x1={i + NA_SPACING}
+                    y1="0"
+                    x2={i + NA_SPACING}
+                    y2="30"
+                    style={{ stroke: colors.gray60, strokeWidth: NA_SPACING }}
```

Comment:
NP: Are these values really tied to the ones below? I mean, hey are the same value, but should they be the same constant?

Reply:
You mean, if it makes sense to have `strokeWidth: NA_SPACING` being tied to `i + NA_SPACING`? If so, yes, it allows us to keep an even pattern of 4px wide strokes, spaced 0px apart.

Reply:
I meant tied to the `rect` and `text` below, but yeah, I see the pattern is actually used by the `rect`, so nevermind :+1: